openquake/hazardlib/sourcewriter.py

```diff
@@ -492,7 +492,7 @@ def build_source_group(source_group):
         attrs['src_interdep'] = source_group.src_interdep
     if source_group.rup_interdep:
         attrs['rup_interdep'] = source_group.rup_interdep
-    if source_group.srcs_weights:
+    if list(source_group.srcs_weights):
         attrs['srcs_weights'] = ' '.join(map(str, source_group.srcs_weights))
```

Comment:
Thinking a bit more, it is probably better to accept `srcs_weights = None` and to avoid writing lots of identical weights.
I will change a bit the code.